packages/integration-tests/projects/suite-web/tests/wallet/add-account-types.test.ts

```diff
@@ -0,0 +1,68 @@
+// @group:suite
+// @retry=2
+
+describe('Add-account-types', () => {
+    beforeEach(() => {
+        cy.task('startEmu', { wipe: true });
+        cy.task('setupEmu', {
+            mnemonic: 'town grace cat forest dress dust trick practice hair survey pupil regular',
+        });
+        cy.task('startBridge');
+
+        cy.viewport(1080, 1440).resetDb();
+        cy.prefixedVisit('/');
+        cy.passThroughInitialRun();
+        cy.discoveryShouldFinish();
+    });
+
+    afterEach(() => {
+        cy.task('stopEmu');
+    });
+
+    /**
+     * Test case
+     * 1. Go to Accounts
+     * 2. Unpack all account types
+     * 3. Get the number of accounts
+     * 4. Create new account for each account type
+     * 5. Get the number of accounts again
+     * 6. Verify that the current number is equeal to previous number + 1
+     */
+    it('Add-account-types-BTC', () => {
+        //
+        // Test preparation
+        //
+        const coin = 'btc';
+        const AccsArray = [
+            ['normal', 'Bitcoin'],
+            ['taproot', 'Bitcoin (Taproot)'],
+            ['segwit', 'Bitcoin (Legacy Segwit)'],
+            ['legacy', 'Bitcoin (Legacy)'],
+        ];
+        //
+        // Test execution
+        //
+        cy.getTestElement('@suite/menu/wallet-index', { timeout: 30000 }).click();
+        cy.getTestElement('@account-menu/arrow').click({ multiple: true });
+
+        AccsArray.forEach(accountarray =>
+            cy
+                .get(
+                    `[type="${accountarray[0]}"] > [data-test^="@account-menu/${coin}/${accountarray[0]}/"]`,
```

Comment:
huh, this is hard to read. are we sure there is no better approach how to write it? preferably simply using data-test attribte and cy.getTestElement?

Reply:
I'm afraid not if we want to keep things generic. I'll add some comments to this parts for more clarity.